modules/promotion/src/CouponAccessControlHandler.php

```diff
@@ -19,9 +19,9 @@ public function checkAccess(EntityInterface $entity, $operation, AccountInterfac
     /** @var \Drupal\Core\Access\AccessResult $result */
     $result = parent::checkAccess($entity, $operation, $account);
 
-    if ($result->isNeutral()) {
+    if ($result->isNeutral() && $promotion = $entity->getPromotion()) {
```

Comment:
If there is no promotion it should always be false, because nothing can be applied.

Reply:
We should explicitly deny.

This is part of the bigger data integrity issues we have.